components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/handler/AbstractEventHandlerTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.event.handler;
+
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.event.IdentityEventConfigBuilder;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.bean.IdentityEventMessageContext;
+import org.wso2.carbon.identity.event.bean.ModuleConfiguration;
+import org.wso2.carbon.identity.event.bean.Subscription;
+import org.wso2.carbon.identity.event.event.Event;
+import java.lang.reflect.Field;
+import java.util.*;
+
+public class AbstractEventHandlerTest {
+
+    List<Subscription> subscriptionList;
+    ModuleConfiguration moduleConfiguration;
+    String moduleName = TestEventHandler.class.getSimpleName();
+    Map<String, ModuleConfiguration> moduleConfigurationMap;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        String home = IdentityEventConfigBuilder.class.getResource("/").getFile();
+        String config = IdentityEventConfigBuilder.class.getResource("/").getFile();
+        System.setProperty("carbon.home", home);
+        System.setProperty("carbon.config.dir.path", config);
+
+        subscriptionList = new ArrayList<>();
+        moduleConfiguration = Mockito.mock(ModuleConfiguration.class);
+        Mockito.doReturn(subscriptionList).when(moduleConfiguration).getSubscriptions();
+
+        IdentityEventConfigBuilder identityEventConfigBuilder = Mockito.mock(IdentityEventConfigBuilder.class);
+        Field field = IdentityEventConfigBuilder.class.getDeclaredField("notificationMgtConfigBuilder");
+        field.setAccessible(true);
+        field.set(null, identityEventConfigBuilder);
```

Comment:
What is the meaning of setting a declared field to a null object?

Reply:
This is since notificationMgtConfigBuilder is a static field and it is not bound to the object. It is bound to the class. So when setting the field, object parameter is sent as null.